lib/sla.js

```diff
@@ -98,46 +99,102 @@ function addChannelsToBridge(inbound, dialed, bridge) {
     answer();
-    defer.resolve(dialed);
-  });
-  dialed.once('ChannelDestroyed', function(event, dialed) {
-    defer.resolve('Dialed channel hungup');
-  });
-  inbound.once('ChannelHangupRequest', function(event, inbound) {
-    var hangup = Q.denodeify(dialed.hangup.bind(dialed));
-    hangup();
+    var toKill = dialed.filter(function (unanswered) {
+      return unanswered.id !== line.id;
+    });
+    toKill.forEach(function (unanswered) {
+      unanswered.removeListener('ChannelDestroyed', dialedChannelHungup);
+      var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+      hangup();
+    });
+    defer.resolve(line);
+  }
+
+  function dialedChannelHungup(event) {
+    numToHangup -= 1;
+    if(numToHangup === 0) {
+      inbound.removeListener('ChannelHangupRequest', inboundHungup);
+      defer.resolve('Dialed channels hungup');
+    }
+  }
+
+  function inboundHungup(event, line) {
+    dialed.forEach(function (unanswered) {
+      unanswered.removeListener('ChannelDestroyed', dialedChannelHungup);
+      var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+      hangup();
+    });
     defer.resolve('Inbound channel hungup');
-  });
+  }
+
+  trunkData.getAllStations()
+    .then(function (stations) {
+      if (stations.length) {
+        stations.forEach(function() {
+          dialed.push(client.Channel());
+        });
+        numToHangup = dialed.length;
+        dialed.forEach(function (channel, i, dialed) {
+          originate.push(Q.denodeify(channel.originate.bind(dialed[i])));
+          originate[i]({endpoint: stations[i].endpoint, app: 'sla',
+            appArgs: 'dialed', timeout: 10})
+            .catch(function (err) {
+              defer.reject(err);
+            });
+          channel.once('StasisStart', dialedEnteredStasis);
+          channel.once('ChannelDestroyed', dialedChannelHungup);
+        });
+        inbound.once('ChannelHangupRequest', inboundHungup);
+      }
+      else {
+        defer.reject(new CustomError('NoStations',
+              'No stations in this trunk.'));
+      }
+    });
   return defer.promise;
 }
 
 /** 
- * Simple utility function for checking if the bridgeName is numerical
- * @param {String} bridgeName - the name of the bridge to be created/used
- * @return {boolean} whether or not the bridgeName is numerical
+ * Utility function for checking if the bridgeName exists in the configuration
+ * file (is valid).
+ * @param {String} configurationFile - the path and filename associated with the
+ * cofiguration file 
+ * @param {String} extension - the name of the extension (and trunk so far)
+ * to be created/used
+ * @return {Q} - Q promise object
  */
-function checkIfNum (string) {
-  if(!isNaN(parseInt(string))) {
-    return true;
-  }
-  else {
-    return false;
-  }
+function getTrunkData (configurationFile, extension) {
+  var defer = Q.defer();
+  dal.getSharedExtension(configurationFile, extension)
```

Comment:
There's no need for a deferred here. Just return dal.getSharedExtension and within it either return a value on success or throw an error if not.

Deferred should only be created explicitly if you are creating a promise supported API or if you want the result of an event listener firing to resolve a promise. Since dal.getSharedExtension is already a promise supported API there's no reason to create a deferred yourself.